pyradiosky/tests/test_skymodel.py

```diff
@@ -965,6 +965,251 @@ def test_polarized_source_smooth_visibilities():
         assert np.all(imag_stokes == 0)
 
 
+@pytest.mark.parametrize(
+    "comp_type, spec_type",
+    [
+        ("point", "subband"),
+        ("point", "spectral_index"),
+        ("point", "flat"),
+        ("healpix", "full"),
+    ],
+)
+def test_concat(comp_type, spec_type, healpix_disk_new):
+    if comp_type == "point":
+        skyobj_full = SkyModel.from_gleam_catalog(GLEAM_vot, spectral_type=spec_type)
+    else:
+        skyobj_full = healpix_disk_new
+
+    # Add on optional parameters
+    skyobj_full.extended_model_group = skyobj_full.name
+    skyobj_full.beam_amp = np.ones((4, skyobj_full.Nfreqs, skyobj_full.Ncomponents))
+
+    skyobj1 = skyobj_full.select(
+        component_inds=np.arange(skyobj_full.Ncomponents // 2), inplace=False
+    )
+    skyobj2 = skyobj_full.select(
+        component_inds=np.arange(skyobj_full.Ncomponents // 2, skyobj_full.Ncomponents),
+        inplace=False,
+    )
+    skyobj_new = skyobj1.concat(skyobj2, inplace=False)
+
+    assert skyobj_new.history != skyobj_full.history
+    expected_history = (
+        skyobj_full.history
+        + " Combined skymodels along the component axis using pyradiosky."
```

Comment:
I think this passing by a fluke?  As noted above, I expect the `history_update_string` to wind up in the middle of the new history.  But the `_combine_histories` method eliminates repeats and, in this case, `skyobj1` and `skyobj2` have identical histories (since they came from the same file) so nothing from the history of `skyobj2` actually winds up in the history of `skyobj_new`.  Or I'm just being dense and missing something entirely.

Reply:
I added testing to cover the new handling which addresses this.